doc/articles/Surfaces.md

```diff
@@ -0,0 +1,256 @@
+
+Issues can happen when trying to retrieve the frame's @ref WGPUTexture, so the application must check @ref WGPUSurfaceTexture `.status` to see if the surface or the device was lost, or some other windowing system issue caused a timeout.
+The environment can also change the surface without breaking it, but making the current configuration suboptimal. 
+In this case, @ref WGPUSurfaceTexture `.suboptimal` will be `true` and the application should (but isn't required to) handle it.
+Surfaces often become suboptimal when the window is resized (so presenting requires resizing a texture, which is both performance overhead, and a visual quality degradation).
+
+This is an example of how to render to a @ref WGPUSurface each frame:
+
+```c
+// Get the texture and handle exceptional cases.
+WGPUSurfaceTexture surfaceTexture;
+wgpuSurfaceGetCurrentTexture(mySurface, &surfaceTexture);
+
+if (surfaceTexture.status != WGPUSurfaceGetCurrentTextureStatus_Success) {
+    // Recover if possible.
+    return;
+}
+if (surfaceTexture.suboptimal) {
+    HandleResize();
+    return;
+}
+
+// The application renders to the texture.
+RenderTo(surfaceTexture.texture);
+
+// Present the texture, it is no longer accessible after that point.
+wgpuSurfacePresent(mySurface);
+```
+
+The behavior of `::wgpuSurfaceGetCurrentTexture``(surface, surfaceTexture)` is:
+
+ - Set `surfaceTexture->texture` to `NULL`.
+ - Set `surfaceTexture->suboptimal` to `false`.
+
+ - If any of these validation steps fails, set `surfaceTexture->status` to `WGPUSurfaceGetCurrentTextureStatus_Error` and return (TODO send error to device?).
+
+   - Validate that `surface` is not an error.
+   - Validate that `surface.config` is not `None`.
+   - Validate that `surface.currentFrame` is `None`.
+
+ - If `surface.config.device` is not alive, set `surfaceTexture->status` to `WGPUSurfaceGetCurrentTextureStatus_DeviceLost` and return.
+ - If the implementation detects any other problem preventing use of the surface, set `surfaceTexture->status` to an appropriate status (other than `Success`, `Error`, or `DeviceLost`) and return.
+ - Let `textureDesc` be `GetSurfaceEquivalentTextureDescriptor(surface.config)`.
+ - Create a new @ref WGPUTexture `t`, as if calling `wgpuDeviceCreateTexture(surface.config.device, &textureDesc)`, but wrapping the appropriate backing resource.
+ - Set `surface.currentFrame` to `t`.
+ - If the implementation detects a reason why the current configuration is suboptimal, set `surfaceTexture->suboptimal` to `true`.
+
+The behavior of `::wgpuSurfacePresent``(surface)` is:
+
+ - If any of these validation steps fails, TODO send error to device?
+
+   - Validate that `surface` is not an error.
+   - Validate that `surface.currentFrame` is not `None`.
+
+ - Do as if `::wgpuTextureDestroy``(surface.currentFrame)` was called.
+ - Present `surface.currentFrame` to the `surface`.
+ - Set `surface.currentFrame` to `None`.
```

Comment:
In other terms, one MUST NOT call `wgpuTextureRelease` on the surface texture returned by `wgpuSurfaceGetCurrentTexture`, right? (unless they've manually added a reference to it) This could be emphasized I think.

Reply:
The opposite actually! I'll make the doc much more clear.